src/analysis/powerset-lattice-impl.h

```diff
@@ -5,61 +5,115 @@
 
 namespace wasm::analysis {
 
-template<size_t N>
-inline BitsetPowersetLattice<N> BitsetPowersetLattice<N>::getBottom() {
+inline FinitePowersetLattice::FinitePowersetLattice(size_t size) : trues(0) {
+  bitvector.reserve(size);
+  for (size_t i = 0; i < size; ++i) {
+    bitvector.push_back(false);
+  }
+}
+
+inline FinitePowersetLattice FinitePowersetLattice::getBottom(size_t size) {
   // Return the empty set as the bottom lattice element.
-  BitsetPowersetLattice<N> result{0};
+  FinitePowersetLattice result(size);
   return result;
 }
 
-template<size_t N>
-inline bool
-BitsetPowersetLattice<N>::isTop(const BitsetPowersetLattice<N>& element) {
+inline bool FinitePowersetLattice::isTop() {
   // Top lattice element is the set containing all possible elements.
-  return element.value.all();
+  return trues == bitvector.size();
 }
 
-template<size_t N>
-inline bool
-BitsetPowersetLattice<N>::isBottom(const BitsetPowersetLattice<N>& element) {
+inline bool FinitePowersetLattice::isBottom() {
   // Bottom lattice element is the empty set.
-  return element.value.none();
+  return trues == 0;
+}
+
+inline void FinitePowersetLattice::set(size_t index, bool value) {
```

Comment:
In a follow-up where we make `FinitePowersetLattice` generic over the set element type `T`, it will be good to make this a private implementation detail and present a more set-like interface.

Reply:
I don't think this applies the same way to the new code structure.